components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/internal/TenantManagementListener.java

```diff
@@ -44,6 +46,7 @@
     public void onTenantCreate(TenantInfoBean tenantInfo) throws StratosException {
         try {
             String tenantDomain = tenantInfo.getTenantDomain();
+            IdentityUtil.threadLocalProperties.get().put(TENANT_NAME_FROM_CONTEXT, tenantDomain);
```

Comment:
Wouldn't it be better to have a direct method to set/clear tenantDomain to/from thread-local variable?

Something like IdentityUtil.setTenantDomainToThreadLocal() and IdentityUtil.clearTenantDomainToThreadLocal()

Reply:
Also, adding a comment on why we need to do this here would be worth to have for future reference.

Reply:
Instead of using the ThreadLocal, initiated a tenant flow here.